lib/WUI/wui_api.cpp

```diff
@@ -347,30 +350,29 @@ uint32_t wui_gcodes_uploaded() {
     return uploaded_gcodes;
 }
 
-void wui_uploaded_gcode(const char *filename, bool start_print) {
+bool wui_uploaded_gcode(const char *filename, bool start_print) {
+    // Note: By checking now and starting it later, we are introducing a short
+    // race condition. Doing it properly would be kind of hard and the risk is
+    // we maybe start the print and don't get the print screen or something
+    // like that ‒ annoying, but not entirely dangerous.
+    //
+    // We assume marlin does another check when asked to print and won't start
+    // a print inside a print or such.
+    //
+    // Also, this introduces another code dependency in the „wrong direction“.
+    // GUI should be a neighbor of WUI and should not depend on each other.
+    // But, well, …
```

Comment:
I agree with your complaints, but the infrastructure is not ready for the task of synchronizing multiple clients among each other.